xds/src/test/java/io/grpc/xds/XdsClientFederationTest.java

```diff
@@ -203,6 +209,19 @@ public void lrsClientsHaveDistinctLoadStatsManagers() throws InterruptedExceptio
     assertThat(loadStatManagers).containsNoDuplicates();
   }
 
+  // Waits for all SynchronizationContext tasks to finish, assuming no new ones get added.
+  private static void waitForSyncContext(SynchronizationContext syncContext)
```

Comment:
I see. 
Maybe you can leverage/abuse  `getSubscribedResourcesMetadataSnapshot()`?  That returns a future.

Reply:
Seems reasonable, I made that change.